### PR TITLE
adding a fuction in DiagoElpa

### PR DESCRIPTION
**Pull request by AcceptedZ**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### What's changed?
adding a fuction in DiagoElpa to provide an option to change mpi_comm_group in elpa_solver, and add a variable in Input.h to get the number as part of the input

### Any changes of core modules? (ignore if not applicable)
changing the mpi_comm_group in elpasolver

### CODE REVIEW

File: source/module_hsolver/diago_elpa.h [hunk 1]
```diff
@@ -15,18 +15,28 @@ namespace hsolver
 
     public:
         void diag(hamilt::Hamilt<T>* phm_in, psi::Psi<T>& psi, Real* eigenvalue_in) override;
+        #ifdef __MPI
+        MPI_Comm setmpicomm();//set mpi comm;
+        void set_comm_num(int cnum);
```
Comment (haozhihan) [inline source/module_hsolver/diago_elpa.h]: Should `setmpicomm` and `set_comm_num` be a **private** func?

**Review by haozhihan — APPROVED** [review]
LGTM

**Comment by Cstandardlib** [comment]
please resolve the conflicts

**Comment by Cstandardlib** [comment]
please resolve the conflicts

**Review by mohanchen — APPROVED** [review]

**Pull request merged**